store/store.go

```diff
@@ -1239,27 +1247,41 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo)
 }
 
 // downloadAndApplyDelta downloads and then applies the delta to the current snap.
-func (s *Store) downloadAndApplyDelta(name string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (path string, err error) {
+func (s *Store) downloadAndApplyDelta(name, targetFn string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (err error) {
 	deltaInfo := &downloadInfo.Deltas[0]
-	workingDir, err := ioutil.TempDir(dirs.SnapPartialBlobDir, "deltas-"+name)
+	// use /var/tmp here because it is not on a tmpfs
+	workingDir, err := ioutil.TempDir("/var/tmp", "deltas-"+name)
```

Comment:
Indeed. It'll also be better to not even create this directory. Using the same logic as above, we can use `targetFn + ".delta.partial"` in the final directory. If this ends up being a bit confusing, let's please just nuke the whole delta logic. This is experimental anyway and we can easily reintroduce on the next release.